cli/src/diff_util.rs

```diff
@@ -167,10 +167,50 @@ fn builtin_diff_format(name: &str, num_context_lines: Option<usize>) -> Option<D
     }
 }
 
-fn default_diff_format(
+fn diff_format_from_settings(
     settings: &UserSettings,
     num_context_lines: Option<usize>,
 ) -> Result<DiffFormat, config::ConfigError> {
+    let config = settings.config();
+    let format = if let Some(args) = config.get("ui.diff-formatter").optional()? {
+        if let CommandNameAndArgs::String(name) = &args {
+            // If setting is a single name starting with a colon it's treated as a built-in format. 
+            // It is still possible to use an external tool starting with a colon passing it as 
+            // an array rather than a string.
+            if name.starts_with(':') {
+              Some(builtin_diff_format(&name[1..], num_context_lines).ok_or_else(||
+                config::ConfigError::Message(format!("Unknown format setting for 'ui.diff-formatter', built-in formats are ':summary', ':types', ':git', ':color-words', and ':stat', or use an external tool")))?)
+                
```

Comment:
nit: `config_error(message).hinted(hint)` ?

Perhaps, we can change the error type to `CommandError`.